src/components/SearchBar.js

```diff
@@ -133,6 +178,86 @@ class SearchBar extends React.Component<SearchBarDefaultProps, SearchBarProps, S
     );
   }
 
+  // queries the index for the saved searches and sets the response to state
+  getSavedSearches() {
+    const username = AuthUtils.getLoggedInUserId();
+    const searcher = this.context.searcher;
+    const { savedSearchWorkflow, savedSearchTable, savedSearchZone } = this.props;
+    if (searcher) {
+      const qr = new SimpleQueryRequest();
+      qr.query = `AND(table:${savedSearchTable}, username_s:"${username}")`;
+      qr.facets = [];
+      qr.restParams.set('zones', [`${savedSearchZone}`]);
+      if (savedSearchWorkflow) {
+        qr.workflow = savedSearchWorkflow;
+      }
+      qr.queryLanguage = 'advanced';
+      searcher.doCustomSearch(qr, (response: QueryResponse | null, error: string | null) => {
+        if (response) {
+          this.setState(
+            {
+              response,
```

Comment:
Could you create a class for the response in the `api directory` and move the formatting you do later on into it to be used here?